lib/src/main/java/graphql/nadel/NextgenEngine.kt

```diff
@@ -343,9 +343,16 @@ internal class NextgenEngine(
             serviceContext = executionContext.getContextForService(service).await(),
             serviceExecutionContext = serviceExecutionContext,
             hydrationDetails = executionHydrationDetails,
-            executableNormalizedField = transformedQuery,
+            // Prefer non __typename field first, otherwise we just get first
+            executableNormalizedField = topLevelFields
+                .asSequence()
+                .filterNot {
+                    it.fieldName == TypeNameMetaFieldDef.name
+                }
+                .firstOrNull() ?: topLevelFields.first(),
```

Comment:
This gets the first non `__typename` field, to maintain the old behavior.

(Refer to above comment).